jekyll.gemspec

```diff
@@ -40,7 +40,7 @@ Gem::Specification.new do |s|
   s.add_runtime_dependency("jekyll-watch",          "~> 2.0")
   s.add_runtime_dependency("kramdown",              "~> 2.3", ">= 2.3.1")
   s.add_runtime_dependency("kramdown-parser-gfm",   "~> 1.0")
-  s.add_runtime_dependency("liquid",                "~> 4.0")
+  s.add_runtime_dependency("liquid",                ">= 4.0", "< 5.4")
```

Comment:
```suggestion
  if RUBY_VERSION.to_f < 3.2.0
    s.add_runtime_dependency("liquid", "~> 4.0.3")
  else
    s.add_runtime_dependency("liquid", "~> 5.4.0")
```

Reply:
I am fine with this as far as you are. But we need to depend on liquid < 5.4.
```
-    s.add_runtime_dependency("liquid", "~> 5.4.0")
+   s.add_runtime_dependency("liquid", "~> 5.3.0")
```

5.4 breaks further as you can see in the the test CI run on my repo: https://github.com/isimluk/jekyll/actions/runs/3777777021/jobs/6421776463 

liquid 5.4.0 changed `Registers` to be a plain old ruby class whilst previously it has been Hash. I have a PR for that at https://github.com/Shopify/liquid/pull/1663 but 5.4.0 remains unusable.

Reply:
> What do you think of just relying on `RUBY_VERSION` to determine the version of liquid to use?

I actually tend to think that is good idea. But how do you feel about the removal of liquid-c support that this PR currently brings ?

Reply:
Yeah setting the latest allowed version to 5.4 sounds reasonable to me. 

Removing liquid-c is outside my understanding. It looks like it was introduced years ago, but wasn't actually being required until https://github.com/jekyll/jekyll/pull/7792 came along. Maybe it's fine to remove